### ISSUE TITLE
Bump Pipeline, K8s and Knative version

### ISSUE THREAD

**Issue by savitaashture** [tektoncd/triggers#1195]
Before doing Triggers v0.16.x release lets bump the Pipeline, K8s and Knative version

**Comment by piyush-garg** [comment]
I tried this and facing some issues like

```
{"level":"info","ts":1629821870.8444512,"logger":"fallback","caller":"eventlistener/eventlistener.go:433","msg":"Created EventListener Deployment el-my-eventlistener in Namespace test-pipelines"}
{"level":"info","ts":1629821870.844536,"logger":"fallback","caller":"eventlistener/eventlistener.go:235","msg":"Created EventListener Service el-my-eventlistener in Namespace test-pipelines"}
--- FAIL: TestReconcile (0.02s)
    --- FAIL: TestReconcile/eventlistener_creation (0.02s)
        logger.go:130: 2021-08-24T21:47:50.843+0530	INFO	eventlistener/controller.go:72	Setting up event handlers
panic: coding error: you must register resource to list kind for every resource you're going to LIST when creating the client.  See NewSimpleDynamicClientWithCustomListKinds or register the list into the scheme: serving.knative.dev/v1, Resource=services out of map[] [recovered]
	panic: coding error: you must register resource to list kind for every resource you're going to LIST when creating the client.  See NewSimpleDynamicClientWithCustomListKinds or register the list into the scheme: serving.knative.dev/v1, Resource=services out of map[]

goroutine 14 [running]:
testing.tRunner.func1.1(0x1b5fa80, 0xc00032e530)
	/usr/lib/golang/src/testing/testing.go:1072 +0x30d
testing.tRunner.func1(0xc000001e00)
	/usr/lib/golang/src/testing/testing.go:1075 +0x41a
panic(0x1b5fa80, 0xc00032e530)
	/usr/lib/golang/src/runtime/panic.go:969 +0x1b9
k8s.io/client-go/dynamic/fake.(*dynamicResourceClient).List(0xc0006bff80, 0x213e360, 0xc00019e010, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/home/pgarg/go/src/github.com/tektoncd/triggers/vendor/k8s.io/client-go/dynamic/fake/simple.go:317 +0xcbf
github.com/tektoncd/triggers/test.GetResourcesFromClients(0xc0000f31e0, 0xc0000f3130, 0xc0000f2000, 0xc0000f26e0, 0xc000689998, 0xc0006c48c0, 0x0, 0x0, 0x1, 0x1, ...)
	/home/pgarg/go/src/github.com/tektoncd/triggers/test/controller.go:385 +0x1775
github.com/tektoncd/triggers/pkg/reconciler/eventlistener.TestReconcile.func40(0xc000001e00)
	/home/pgarg/go/src/github.com/tektoncd/triggers/pkg/reconciler/eventlistener/eventlistener_test.go:1501 +0x2e5
testing.tRunner(0xc000001e00, 0xc000452a90)
	/usr/lib/golang/src/testing/testing.go:1123 +0xef
created by testing.(*T).Run
	/usr/lib/golang/src/testing/testing.go:1168 +0x2b3
FAIL	github.com/tektoncd/triggers/pkg/reconciler/eventlistener	0.076s

```

When I am using another client like NewSimpleClient from client-go, facing other issues like

```
--- FAIL: TestGetResourcesFromClients (0.00s)
    --- FAIL: TestGetResourcesFromClients/empty (0.00s)
        logger.go:130: 2021-08-24T21:06:24.293+0530	PANIC	dynamicclient/dynamicclient.go:45	Unable to fetch k8s.io/client-go/dynamic.Interface from context.
panic: Unable to fetch k8s.io/client-go/dynamic.Interface from context. [recovered]
	panic: Unable to fetch k8s.io/client-go/dynamic.Interface from context.

```

**Comment by piyush-garg** [comment]
There are two PRs now #1196 and #1209 

I raised #1196  with the help of @mattmoor only and the reason I was working on this was having pipeline v0.27.1 and have the same set of dep like knative and k8s as in pipeline for v0.16.x so that when we bump the deps in CLI for pipeline and triggers, everything becomes consistent in terms of indirect deps (we faced difficulties in past) as we are working to have v1beta1 support for resources along with pipeline v0.27.1

I am fine if we want to go with the other PR #1209 and let's do a release as I remember this was the only thing pending for release. TIA

cc @dibyom @mattmoor @savitaashture @vdemeester

**Comment by piyush-garg** [comment]
Fixed by #1196 

/close
